provider/terraform.rb

```diff
@@ -125,14 +125,35 @@ def titlelize_property(property)
       p
     end
 
+    # Compile files that aren't product specific like provider.go
+    def compile_provider_files(output_folder, products, version)
```

Comment:
Ideally, I think this function would be combined with [`compile_common_files`](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/provider/core.rb#L246-L253) and would pick up on common files automatically, same as we do for common copied / compiled files. Adding files 1-by-1 is annoying as we've seen with the validator, and especially easy to miss for developers not on our team.